benchopt/runner.py

```diff
@@ -205,6 +205,65 @@ def get_next(x): return max(x / rho, MIN_TOL)
     return curve, status
 
 
+def get_callback(objective, max_iter, deadline,  meta):
+    """Get callback function and store informations for the curve.
+
+    Parameters
+    ----------
+    objective : instance of BaseObjective
+        The objective to minimize.
+    maxiter : maximum number of iterations in the run.
+    deadline : maximum time allowed to perform the run.
```

Comment:
```suggestion
    max_iter : int
        Maximum number of iterations to run for.
    deadline : float
        Deadline after which to stop the computations. This will be
        used to respect the timeout for each solver.
```